### ISSUE TITLE
[ISSUE]: Azure DevOps Pipeline GitVersion Task with error: "maxBuffer length exceeded"

### ISSUE THREAD

**Issue by ulluoink** [GitTools/actions#1237]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched [issues](https://github.com/GitTools/actions/issues) to ensure it has not already been reported

### GitVersion package

AzureDevops task

### What are you seeing?

In Azure yml pipeline step we use the gitversion tasks just like in the usage examples and get following error when executing 

gitversion/execute@3.0.0

This is the output: 

Command: git -C C:/a/15/s rev-parse --is-shallow-repository
Command: C:\hostedtoolcache\windows\GitVersion.Tool\6.0.2\dotnet-gitversion.exe C:/a/15/s /output json /output buildserver /config C:\a\15\s\GitVersion.yml
##[error]stdout maxBuffer length exceeded

### What is expected?

well guess what ? should work 

### Steps to Reproduce

azure devops pipeline with these steps:

steps:
- task: gitversion/setup@3.0.0
  displayName: 'Install GitVersion'
  inputs:
    versionSpec: '6.x'

- task: gitversion/execute@3.0.0
  displayName: 'Configure Git Version'
  name: ${{ parameters.VersionStepName }}
  inputs:
    useConfigFile: true
    configFilePath: ${{ parameters.ConfigFilePath }}

**Comment by edwardvandevorst-philips** [comment]
I see similar issue on GitHub. Removing stale branches, i.e. making sure that the output is smaller, solves the issue.

**Comment by arielangeles** [comment]
Same here. This happens in task version 2.x and 3.x, using version spec 5.x. Workaround for now is to go back to v1.x on those pipelines.

**Comment by LiamCottrell** [comment]
Had the same issue as well as others, version 3 of task with version spec 5 of the tool. Couldn't resolve until i went back to version 0 of the devops tasks.

**Comment by luk355** [comment]
This issue is still occurring on our repo with v3 of the execute task. The version calculation works fine with v0.




```yaml
- task: gitversion/setup@3.0.0
  displayName: Install GitVersion
  inputs:
    versionSpec: '6.0.4'

## fails
- task: gitversion/execute@3.0.0
  displayName: Determine Version
  inputs:
    useConfigFile: true
    configFilePath: 'GitVersion.yml'

## works
- task: gitversion/execute@0
  displayName: Determine Version
  inputs:
    useConfigFile: true
    configFilePath: 'GitVersion.yml'
```

**Comment by arturcic** [comment]
@luk355 can you attach some logs as it's not clear what you see

**Comment by luk355** [comment]
@arturcic, thanks for prompt response. I've updated the post above with pipeline output. It fails with `##[error]stdout maxBuffer length exceeded`. Let me know if you require more info.

**Comment by arturcic** [comment]
> [@luk355](https://github.com/luk355) this is the buffer size
> 
> [actions/src/agents/common/build-agent.ts](https://github.com/GitTools/actions/blob/8bbeaf6ace5bb834365a9490b42482a6b0a64f4f/src/agents/common/build-agent.ts#L233)
> 
> Line 233 in [8bbeaf6](/GitTools/actions/commit/8bbeaf6ace5bb834365a9490b42482a6b0a64f4f)
> 
>  const commandOptions: ExecOptions = { maxBuffer: 1024 * 1024 * 10 } // 10MB 
> . I wonder if your log output is bigger then 10MB. Mind if you pull the same branch you run in the Azure Pipelines and run this command in the git root and see how big the log is?
> export TF_BUILD=true
> export GIT_BRANCH={your-branch-name}
> dotnet-gitversion /home/vsts/work/1/s /output json /output buildserver /config GitVersion.yml

you will need to install the dotnet sdk if you don't have it

**Comment by luk355** [comment]
I did try to run the command locally on on one of the failing branches in Azure Pipelines.

```
dotnet-gitversion . /output json /output buildserver /config GitVersion.yml > gitversion.log
```

... when checking the size of `gitversion.log` file, it has 150076 bytes ~ 0.143 MB

I thought the git state (branch) before the command is run might have affect on the log output but it seems it does not. I have tried to run this command on `main` branch as well as on a detached state based on the commit (the same way as a repo is checked out in Azure Pipelines) and the log output size is very similar.
